src/main/java/uk/gov/register/views/PreviewPageView.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.register.views;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.register.configuration.HomepageContent;
+import uk.gov.register.configuration.RegisterTrackingConfiguration;
+import uk.gov.register.core.RegisterName;
+import uk.gov.register.core.RegisterReadOnly;
+import uk.gov.register.core.RegisterResolver;
+import uk.gov.register.resources.RequestContext;
+import uk.gov.register.thymeleaf.ThymeleafView;
+import uk.gov.register.util.ElementType;
+import uk.gov.register.views.representations.ExtraMediaType;
+
+import javax.inject.Provider;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+
+public abstract class PreviewPageView extends ThymeleafView {
+
+    private static final String NUMBER_FORMAT = "#,###";
+    private static final String PREVIEW_HTML_PAGE = "preview.html";
+
+    protected static final String YAML = "YAML";
+    protected static final String TTL = "TTL";
+    protected static final String JSON = "JSON";
+
+    private final String key;
+    private final String previewType;
+    private final HomepageContent homepageContent;
+    private final Provider<RegisterName> registerNameProvider;
+    private final ElementType elementType;
+    private final int totalRecords;
+    private final int totalEntries;
+
+    public PreviewPageView(final RequestContext requestContext,
+                           final RegisterTrackingConfiguration registerTrackingConfiguration,
+                           final RegisterResolver registerResolver,
+                           final RegisterReadOnly register,
+                           final String key,
+                           final String previewType,
+                           final ElementType elementType,
+                           final HomepageContent homepageContent,
+                           final Provider<RegisterName> registerNameProvider) {
+        super(requestContext, PREVIEW_HTML_PAGE, registerTrackingConfiguration, registerResolver, register);
+        this.key = key;
+        this.previewType = previewType;
+        this.elementType = elementType;
+        this.homepageContent = homepageContent;
+        this.registerNameProvider = registerNameProvider;
+        totalRecords = register.getTotalRecords();
```

Comment:
Could we just call these methods down below in the `getTotalRecords` and `getTotalEntries` methods rather than creating new properties for them?

Reply:
Actaully, no. At least without modifiying the class `ThymeleafView` and exposing the variable `registers` what at this point looks more invasive than just store the values in the class it needs them. In addtion, there is a naming problem if this is done because the `getRegister` is not returning the `register` object, it's returning the `RegisterMetadata` object.

Reply:
Discussed.